pkg/services/ngalert/api.go

```diff
@@ -148,36 +111,48 @@ func (ng *AlertNG) deleteAlertDefinitionEndpoint(c *models.ReqContext) api.Respo
 // updateAlertDefinitionEndpoint handles PUT /api/alert-definitions/:alertDefinitionId.
 func (ng *AlertNG) updateAlertDefinitionEndpoint(c *models.ReqContext, cmd updateAlertDefinitionCommand) api.Response {
 	cmd.ID = c.ParamsInt64(":alertDefinitionId")
```

Comment:
Need to set the OrgID from signed in user here. Currently going through as zero.

Reply:
Also needed on 84 above I think in getAlertDefinitionEndpoint

Reply:
I think it's better to fix this inside `updateAlertDefinition()` by setting OrgID similar to what we do for the rest fields.
https://github.com/grafana/grafana/blob/ec7943eed942bc888c737b45c487ffc0c850b586/pkg/services/ngalert/database.go#L112-L117

If the `cmd.OrgID` is zero then[ the specific field is not updated](https://godoc.org/github.com/go-xorm/xorm#Engine.Update).  WDYT?

Reply:
It makes sense that a user can not change the OrgID of an alert def (I think). But we do want to make sure that they can't update or get an alert def that is not in their organization.